scripts/run_lighthouse_tests.py

```diff
@@ -50,11 +53,94 @@ def run_lighthouse_checks():
         python_utils.PRINT(line[:-1])
 
 
-def main():
-    """Runs lighthouse checks and deletes reports."""
-    setup_and_install_dependencies()
+def start_google_app_engine_server():
+    """Start the Google App Engine server."""
+    app_yaml_filepath = 'app.yaml'
+
+    python_utils.PRINT('Starting oppia server...')
+
+    p = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' % (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_HOME,
+            APP_ENGINE_PORT, app_yaml_filepath), shell=True)
+    RUNNING_PROCESSES.append(p)
+
+
+def download_and_install_nginx():
+    """Download and install nginx."""
+    python_utils.PRINT('Installing nginx...')
+    update_command = ['sudo', 'apt', 'update']
+    install_command = ['sudo', 'apt', 'install', 'nginx']
+
+    subprocess.check_call(update_command)
+    subprocess.check_call(install_command)
+
+
+def start_proxy_server():
+    """Start the nginx proxy server."""
+    filepath = os.path.join(common.CURR_DIR, 'nginx.conf')
+
+    python_utils.PRINT('Starting proxy server...')
+    start_server_command = ['sudo', 'nginx', '-c', filepath]
+    subprocess.Popen(start_server_command)
+
+
+def run_lighthouse_checks_with_compression():
+    """Run lighthouse checks with compression enabled."""
+    try:
+        python_utils.PRINT('Checking if nginx is installed...')
+        check_nginx_command = ['which', 'nginx']
+        subprocess.check_call(check_nginx_command)
+    except subprocess.CalledProcessError:
```

Comment:
I don't see anything related to compression here, can you explain what's going on?